packages/i18next/src/lib/functions.ts

```diff
@@ -198,17 +273,77 @@ export async function editLocalized<TKeys extends TFunctionKeys = string, TInter
+ * @since 2.4.0
+ * @param target The interaction to editReply.
+ * @param options The language keys to be sent.
+ * @example
+ * ```typescript
+ * // Using a string to specify the key to send
+ * await editLocalized(interaction, 'commands/ping:fail');
+ * // ➡ "Pong!"
+ * ```
+ */
+export async function editLocalized<TKeys extends TFunctionKeys = string>(
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	keys: TKeys | TKeys[]
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']>>;
+/**
+ * Edits a deferred/replied interaction using an objects option.
+ * @since 2.4.0
+ * @param target The interaction to editReply.
+ * @param options The options to be sent, requiring at least `keys` to be passed.
+ * @example
+ * ```typescript
+ * // Using an object to specify the key to send
+ * await editLocalized(interaction, { keys: 'commands/ping:fail' });
+ * // ➡ "Pong!"
+ * ```
+ * @example
+ * ```typescript
+ * // Passing interpolation options into i18next
+ * const latency = 42;
+ * const took = 96;
+ *
+ * await editLocalized(interaction, {
+ * 	keys: 'commands/ping:success',
+ * 	formatOptions: { latency, took }
+ * });
+ * // ➡ "Pong! Took me 96ms to reply, and my heart took 42ms to beat!"
+ * ```
+ */
 export async function editLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
-	target: Message,
-	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap>
-): Promise<Message> {
+	target: BaseCommandInteraction | MessageComponentInteraction,
+	options: LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']>>;
+export async function editLocalized<TKeys extends TFunctionKeys = string, TInterpolationMap extends NonNullObject = StringMap>(
+	target: BaseCommandInteraction | Message | MessageComponentInteraction,
+	options: TKeys | TKeys[] | LocalizedMessageOptions<TKeys, TInterpolationMap> | LocalizedInteractionReplyOptions<TKeys, TInterpolationMap>
+): Promise<ReturnType<(BaseCommandInteraction | MessageComponentInteraction)['editReply']> | Message> {
+	// Handle Interactions:
+	if (target instanceof BaseCommandInteraction || target instanceof MessageComponentInteraction) {
+		return target.editReply(await resolveOverloads(target, options));
+	}
+
+	// Handle Message:
 	return target.edit(await resolveOverloads(target, options));
 }
 
 /**
  * @internal
  */
 async function resolveLanguage(context: InternationalizationContext): Promise<string> {
+	Object.defineProperty(context, 'author', {
+		get: () => {
+			process.emitWarning(
+				"InternationalizationContext's `author` property is deprecated and will be removed in the next major version. Please use `InternationalizationContext.user` instead.",
+				'DeprecationWarning'
+			);
+			return context.user;
+		}
+	});
```

Comment:
Missing setter, the property is readonly at the moment, which makes it breaking.

And I guess it needs to be configurable and writable?